staging/src/k8s.io/controller-manager/pkg/leadermigration/config/config.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	util "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	internal "k8s.io/controller-manager/config"
+	"k8s.io/controller-manager/config/v1alpha1"
+	"sigs.k8s.io/yaml"
+)
+
+// ResourceLockLeases is the resourceLock value for 'leases' API
+const ResourceLockLeases = "leases"
+
+// ResourceLockEndpoints is the resourceLock value for 'endpoints' API
+const ResourceLockEndpoints = "endpoints"
+
+var cfgScheme = runtime.NewScheme()
+
+func init() {
+	// internal
+	util.Must(internal.AddToScheme(cfgScheme))
+
+	// v1alpha1
+	util.Must(v1alpha1.AddToScheme(cfgScheme))
+	util.Must(cfgScheme.SetVersionPriority(v1alpha1.SchemeGroupVersion))
+}
+
+// ReadLeaderMigrationConfiguration reads LeaderMigrationConfiguration from a YAML file at the given path.
+// The parsed LeaderMigrationConfiguration may be invalid.
+// It returns an error if the file did not exist.
+func ReadLeaderMigrationConfiguration(configFilePath string) (*internal.LeaderMigrationConfiguration, error) {
+	data, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read leader migration configuration from %q [%v]", configFilePath, err)
+	}
+	var decodedConfig v1alpha1.LeaderMigrationConfiguration
+	err = yaml.Unmarshal(data, &decodedConfig)
```

Comment:
Are there any benefits to using the `UniversalDecoder().Decode()`?  For example: 

```
	configObj, gvk, err := codecs.UniversalDecoder().Decode(data, nil, nil)
	if err != nil {
		return nil, err
	}
```

Reply:
(See [this usage](https://github.com/kubernetes/kubernetes/blob/c084d57b18a7c90c14bc13dc2daa256e84037a74/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L208-L225), I think its similar)

Reply:
Thank you! That's a very good point! I can get parsing, conversion, and type checking for free.